quarry/utils/buffer.py

```diff
@@ -52,13 +82,13 @@ def discard(self):
 
         self.pos = len(self.buff)
 
-    def read(self, length=None):
+    def read(self, length=0):
         """
         Read *length* bytes from the beginning of the buffer buffer, or all
-        bytes if *length* is ``None``
+        bytes if *length* is ``0``
         """
 
-        if length is None:
+        if (length == 0):
```

Comment:
This is a stylistic judgement and pretty subjective. Using parentheses for `if` is very uncommon.